include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,430 @@
+     *
+     * \returns \c std::string The stored string.
+     * \par Exceptions
+     * Strong exception guarantee. No data is modified.
+     */
+    std::string string() const
+    {
+        return std::string{ _lit.cbegin(), _lit.cend() - 1};
+    }
+
+    /*!\brief Returns the content represented as 0-terminated c-style string.
+     *
+     * \returns `const char *` The stored string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const char * c_str() const noexcept
+    {
+        return _lit.data();
+    }
+    //!\}
+
+    //!\brief Implicit string conversion.
+    operator std::string() const
+    {
+        return this->string();
+    }
+};
+
+/*!\brief Swaps two constexpr_strings.
+ *
+ * \par Excecptions
+ * noexcept if `lhs.swap(rhs)` is `noexcept`. Otherwise it holds strong exception guarantee.
+ */
+template <std::size_t N>
+void swap(constexpr_string<N> & lhs, constexpr_string<N> & rhs) noexcept(noexcept(lhs.swap(rhs)))
+{
+    lhs.swap(rhs);
+}
+
+/*!\name Deduction guides
+ * \{
+ */
+//!\brief Deduces constexpr_string from string literals.
+template <std::size_t N>
+constexpr_string(const char (&)[N]) -> constexpr_string<N - 1>;
+
+//!\brief Deduces constexpr_string from std::array of type char.
+template <std::size_t N>
+constexpr_string(std::array<char, N> const &) -> constexpr_string<N>;
+
+//!\brief Deduces constexpr_string from char.
+constexpr_string(const char) -> constexpr_string<1>;
+//!\}
+
+/*!\brief Helper function to construct the seqan3::detail::constexpr_string from c-style string.
+ * \tparam Is Parameter pack over all indices of extractable characters.
+ * \param c_str The c-style string to extract characters from.
+ * \see make_constexpr_string
+ * \return An seqan3::detail::constexpr_string of size `sizeof...(Is)` containing the respective part of `c_str`.
+ * \par Exceptions
+ * No-throw guarantee.
+ * \par Thread-Safety
+ * Thread safe.
+ * \ingroup container
+ */
+template <std::size_t ...Is>
+constexpr auto
+make_constexpr_string_impl(const char * const c_str, std::index_sequence<Is...> const & /*index_seq*/) noexcept
+{
+    return constexpr_string{ std::array{c_str[Is]...} };
+}
+
+/*!\brief Extracts `N-1` consecutive characters from a `const char *` and returns them as a seqan3::detail::constexpr_string.
+ * \tparam N The length of the c-style string.
+ * \param c_str The string literal to extract characters from.
+ *
+ * The function assumes that `c_str[N]` is the pass-the-end symbol not included in the seqan3::detail::constexpr_string.
+ * \return An seqan3::detail::constexpr_string of size `N-1` containing the respective part of `c_str`.
+ * \par Exceptions
+ * No-throw guarantee.
+ * \par Thread-Safety
+ * Thread safe.
+ * \ingroup container
+ */
+template <std::size_t N>
+constexpr auto
+make_constexpr_string(const char * const c_str) noexcept
+{
+    return make_constexpr_string_impl(c_str, std::make_index_sequence<N - 1>{});
+}
```

Comment:
Is this function used frequently at all? If not maybe move it to detail?

Reply:
Actually after redesigning the reflection, we don't need it anymore.